### DISCUSSION TITLE
Cut off legacy ASP.NET (full .NET) support?

### DISCUSSION THREAD

**Discussion by RicoSuter** [RicoSuter/NSwag#3760] [Q&A]
I'm working on NSwag v14 and NJsonSchema v11. 
Thinking about removing full .NET / classic ASP.NET support...
This would make the project much simpler, and a lot of old code can be removed and does not have to be maintained anymore.

**Comment by knoepdan** [comment]
I thinks at some point soon that makes sense. (However, I would like to have see a few minor things still being supported in the old .Net framework. Will try to create pull requests for this myself). Maybe you could also get rid of older .Net core version. 

If you drop the support, it might make sense to split the project it's a easily recognizable that there is a NSWAG version that supports the old .net framework and it would be possible to add bugfixes.

**Comment by olenitsj** [comment]
I am afraid allot of enterprise solutions are still on asp.net framework. Cutting it off right now, and not supporting v13 with bug/security fixes would be very very problematic for many. 

See: https://survey.stackoverflow.co/2022/#section-most-popular-technologies-web-frameworks-and-technologies

Asp.net core 18.59%
Asp.net (old) 14.9%

This means that 44.4% of ASP.NET solutions worked with daily, are still old asp.net framework. Killing it of right now, would mean disappointing just a little less than half of the population.

On the other hand, killing it in v14 and supporting v13 with security/bug fixes for some time, would push them to move to .net core in the near future, but this needs to be done "softly". As many of us are still fighting the legacy monsters.

**Comment by shanji97** [comment]
Since our solution still uses ASP .NET 4.8, my question is what documentation (and version) to look for just to start out. Can I still use the newest Nuget package(s)?

One suggestion I have is to cut off Windows Phone 8.1 first (don't get me wrong, it was a great system).
